From 84ef35a33cc5bd754118fc785659e071d5c481f3 Mon Sep 17 00:00:00 2001
From: Jason Kridner <jdk@ti.com>
Date: Fri, 19 Sep 2014 03:49:43 +0000
Subject: [PATCH 3/4] pru_speak: fix dma mask

Hint from kumar.abhishek.kakkar@gmail.com who also informed me that
Deepak Karki did start the work of doing the integration to the shared
driver at https://github.com/deepakkarki/pruspeak/tree/pru_rproc_ps/src/driver.
---
 drivers/remoteproc/pru_speak.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/remoteproc/pru_speak.c b/drivers/remoteproc/pru_speak.c
index bbee465..307525c 100644
--- a/drivers/remoteproc/pru_speak.c
+++ b/drivers/remoteproc/pru_speak.c
@@ -341,6 +341,7 @@ static int pru_speak_probe(struct platform_device *pdev)
 	pp->shm_code.size_in_pages = PS_SHM_CODE_SIZE;
 	pp->shm_code.idx = PS_SHM_CODE_IDX;
 	pp->shm_code.is_valid = 1;
+	dma_set_coherent_mask(dev, 0xFFFFFFFF);
 	pp->shm_code.vaddr = dma_zalloc_coherent(dev, pp->shm_code.size_in_pages * PAGE_SIZE,
 							(dma_addr_t *) &(pp->shm_code.paddr), GFP_DMA);
 	if(!(pp->shm_code.vaddr)){
-- 
2.1.0

